docs/guides/spa-mode.md

```diff
@@ -0,0 +1,96 @@
+---
+title: SPA Mode
+---
+
+# SPA Mode
+
+From the beginning, Remix's opinion has always been that you own your server architecture. This is why Remix is built on top of the [Web Fetch API][fetch] and can run on any modern [runtime][runtimes] via built-in (or community-provided) adapters. While we believe that having a server provides the best UX/Performance/SEO/etc. for _most_ apps, it is also undeniable that there exist plenty of valid use cases for a Single Page Application in the real world:
+
+- You prefer to deploy your app via static files on Github Pages or another CDN
+- You don't want to manage a server, or run a Node.js server
+- You're developing a special type of embedded app that can't be server rendered
+- "Your boss couldn't care less about the UX ceiling of SPA architecture and won't give your dev teams time/capacity to re-architect things" [- Kent C. Dodds][kent-tweet]
+
+That's why we added support for **SPA Mode** in [2.5.0][2.5.0] (per this [RFC][rfc]), which builds heavily on top of the [Client Data][client-data] APIs.
+
+## What is SPA Mode?
+
+SPA Mode is basically what you'd get if you had your own [React Router + Vite][rr-setup] setup using `createBrowserRouter`/`RouterProvider`, but along with some extra Remix goodies:
+
+- File-based routing (or config-based via [`routes()`][routes-config])
+- Automatic route-based code-spitting via [`route.lazy`][route-lazy]
```

Comment:
So every route is automatically turned into a separate file in the Vite output? Any way to opt out of this? My [main work app](https://github.com/oxidecomputer/console) (a Vite+RR app I'm excited to try SPA mode on) has one main bundle and two secondary ones for big libraries, and I like that!

```
dist/assets/Terminal-wrK8jj1j.js                            287.61 kB │ gzip:  71.75 kB │ map:   590.14 kB
dist/assets/TimeSeriesChart-wrSD895q.js                     370.29 kB │ gzip: 103.16 kB │ map: 1,646.72 kB
dist/assets/app-DtIhrgNj.js                               1,038.45 kB │ gzip: 323.32 kB │ map: 4,913.94 kB
```

Reply:
Curious - what's the benefit?

Reply:
Not at the moment - that's just how the Remix manifest works under the hood.  I don't know enough of the vite configs though to know if you have any more granular control over the chunking logic. 

You don't mind the cost of your larger-than-neccesary startup bundle?  If the goal is quicker client-side navs, then I'd reach for `<Link prefetch="render">`

Reply:
That asset is downloaded once and cached in the browser, so no matter what page they hit after that (except for 3 or 4 pages that use the extra chunks) there is nothing to download. I figured in practice the extra transfer time on uncached pageload would not be the dominant factor in perceived performance, but now that I'm checking it, you might be right and it would be better to DL less up front. I'll give SPA mode a try either way as soon as it's out!

Reply:
Another point is larger bundles invalidate more often - so you could potentially get better caching with smaller bundles.

Reply:
Another very interesting point, thank you. It hadn’t occurred to me the hash filenames might _ever_ be the same across releases (my one big file always changes its hash, of course). I’ll have to test that out.